contracts/voting/dao-voting-cw721-staked/src/contract.rs

```diff
@@ -89,10 +89,31 @@ pub fn instantiate(
         NftContract::New {
             code_id,
             label,
-            name,
-            symbol,
+            msg: instantiate_msg,
             initial_nfts,
         } => {
+            // The instantiate message for the NFT contract is likely to be a cw721 or sg721
+            // instantiate message. We need to deserialize it and update the minter to be this
+            // contract.
+            //
+            // Custom NFT contracts with Instantiate messages that do fit the cw721 or sg721
+            // interface will not work with this contract. Use NftContract::Existing instead.
+            let cw721_instantiate_msg = from_binary::<cw721_base::InstantiateMsg>(&instantiate_msg);
+            let sg721_instantiate_msg = from_binary::<sg721::InstantiateMsg>(&instantiate_msg);
+
+            // Update the minter to be this contract, reserialize into Binary
+            let instantiate_msg = match (cw721_instantiate_msg, sg721_instantiate_msg) {
```

Comment:
feels like having some sort of structure here could be nice and would help keep the key contract methods short and concise. what do you think?
didn't test but i'm thinking something along the lines of:
```rust
enum NftInstantiateMsg {
    Cw721(cw721_base::InstantiateMsg),
    Sg721(sg721::InstantiateMsg),
    // any future 721 implementations?
}

impl NftInstantiateMsg {
    fn update_minter(&mut self, minter: &str) {
        match self {
            NftInstantiateMsg::Cw721(msg) => msg.minter = minter.to_string(),
            NftInstantiateMsg::Sg721(msg) => msg.minter = minter.to_string(),
        }
    }
}

fn try_deserialize(instantiate_msg: Binary) -> Result<NftInstantiateMsg, ContractError> {
    if let Ok(cw721_msg) = from_binary::<cw721_base::InstantiateMsg>(&instantiate_msg) {
        return Ok(NftInstantiateMsg::Cw721(cw721_msg));
    }

    if let Ok(sg721_msg) = from_binary::<sg721::InstantiateMsg>(&instantiate_msg) {
        return Ok(NftInstantiateMsg::Sg721(sg721_msg));
    }
    
    Err(ContractError::NftInstantiateError {})
}

```

Reply:
Oh, love this.